src/openag_brain/software_modules/recipe_handler.py

```diff
@@ -105,6 +108,43 @@ def __iter__(self):
             yield (time.time(), variable, value)
         yield (time.time(), RECIPE_END.name, self.id)
 
+def hrs_to_seconds(hrs):
+    return hrs * (60 * 60)
+
+def days_to_seconds(days):
+    return hrs_to_seconds(days * 24)
+
+@interpret_recipe.register("phased")
+class PhasedRecipeInterpreter:
+    def __init__(self, recipe, start_time=None, timeout=1):
+        init_time = rospy.get_time()
+        start_time = start_time or init_time
+        if start_time > init_time:
+            raise ValueError("Recipes cannot be scheduled for the future")
+        self.start_time = start_time
+        self.stages = recipe["operations"]
+        self.id = recipe["_id"]
+        self.timeout = timeout
+
+    def __iter__(self):
+        yield (rospy.get_time(), RECIPE_START.name, self.id)
+        for stage in self.stages:
+            stage_duration = days_to_seconds(stage["days"])
+            end_of_stage = (rospy.get_time() - self.start_time) + stage_duration
```

Comment:
I personally think it's much easier to reason about stages as a number of day-night cycles. And if, as we often claim, plants actually grow faster in our system than with other conventional methods, directly transcribing "farmer's knowledge" isn't particularly useful anyway.

Another way to handle this if we keep the current semantics is to have the system basically "check your math" and round the `days` field to the nearest value that is a multiple of the length of the day-night cycle.